test/testhelpers/ULPError.jl

```diff
@@ -0,0 +1,57 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+module ULPError
+    export ulp_error, ulp_error_maximum
+    @noinline function throw_invalid()
+        throw(ArgumentError("invalid"))
+    end
+    function ulp_error(accurate::AbstractFloat, approximate::AbstractFloat)
+        if isnan(accurate)
+            @noinline throw_invalid()
+        end
+        if isnan(approximate)
+            @noinline throw_invalid()
+        end
```

Comment:
this should return `Inf` if one is NaN and the other isn't, or 0 if both are NaN.